packages/application-components/src/components/modal-page/modal-page.js

```diff
@@ -0,0 +1,163 @@
+    default:
+      return baseZIndex;
+  }
+};
+
+const getColumnSpacing = level => {
+  const baseSpacing = 64;
+  switch (level) {
+    case 'two':
+      return `${baseSpacing}px ${baseSpacing}px auto`;
+    case 'three':
+      return `${baseSpacing}px ${baseSpacing}px ${baseSpacing}px auto`;
+    default:
+      return `${baseSpacing}px auto`;
+  }
+};
+
+const getStyles = props => ({
+  overlay: {
+    zIndex: props.zIndex || getZIndexForLevel(props.baseZIndex, props.level),
+  },
+  content: {
+    gridTemplateColumns: getColumnSpacing(props.level),
+  },
+});
+
+const Spacers = props => {
+  switch (props.level) {
+    case 'two':
+      return (
+        <React.Fragment>
+          <div />
+          <div />
+        </React.Fragment>
+      );
+    case 'three':
+      return (
+        <React.Fragment>
+          <div />
+          <div />
+          <div />
+        </React.Fragment>
+      );
+    default:
+      return <div />;
+  }
+};
+Spacers.displayName = 'Spacers';
+Spacers.propTypes = {
+  level: PropTypes.oneOf(['one', 'two', 'three']).isRequired,
+};
+
+const ModalPage = props => (
+  <Modal
+    isOpen={props.isOpen}
+    onRequestClose={props.onClose}
+    shouldCloseOnOverlayClick={Boolean(props.onClose)}
+    shouldCloseOnEsc={Boolean(props.onClose)}
+    overlayClassName={styles['modal-overlay']}
+    className={styles['modal-content']}
+    contentLabel={props.title}
+    parentSelector={parentSelector}
+    ariaHideApp={false}
+    style={getStyles(props)}
+  >
+    <Spacers level={props.level} />
+    <div className={styles['page-container']}>
+      <Card className={styles['page-card']}>
+        <Spacings.Stack scale="m">
+          <div className={styles['page-header']}>
+            <Spacings.Inline
+              scale="m"
+              alignItems="center"
+              justifyContent="space-between"
+            >
+              <Text.Headline
+                elementType="h3"
+                truncate={true}
+                title={props.title}
+              >
+                {props.title}
+              </Text.Headline>
+              {props.onClose && (
+                <SecondaryIconButton
+                  label="Close page"
+                  onClick={props.onClose}
+                  icon={<CloseIcon size="medium" />}
+                />
+              )}
+            </Spacings.Inline>
+            <Spacings.Inline
+              scale="m"
+              alignItems="center"
+              justifyContent="space-between"
+            >
+              {props.subtitle}
+              <div>{props.components.actions}</div>
+            </Spacings.Inline>
+          </div>
+          <div className={styles['page-content']}>{props.children}</div>
+        </Spacings.Stack>
+      </Card>
+    </div>
+  </Modal>
+);
+ModalPage.displayName = 'ModalPage';
+ModalPage.propTypes = {
+  isOpen: PropTypes.bool.isRequired,
+  onClose: PropTypes.func,
+  zIndex: PropTypes.number,
+  baseZIndex: PropTypes.number,
```

Comment:
Why base and normal? Would one be enough and we just add to that. What would e.g. happen if base would be bigger the normal index?